lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2254,16 +2254,19 @@ ParamSpecifierRequest::evaluate(Evaluator &evaluator,
   if (auto transferring = dyn_cast<TransferringTypeRepr>(nestedRepr)) {
     // If we do not have an Ownership Repr, return implicit copyable consuming.
     auto *base = transferring->getBase();
-    if (!isa<OwnershipTypeRepr>(base)) {
+    if (!param->getInterfaceType()->isNoEscape() &&
```

Comment:
Could you please double-check that this won't cause request dependency cycles with InterfaceTypeRequest?

Reply:
I checked. I didn't see anything obvious.